consensus/hotstuff/verification/common.go

```diff
@@ -68,24 +69,42 @@ func checkVotesValidity(votes []*model.Vote) error {
 	return nil
 }
 
+type aggregate struct {
+	lastStakingSigners map[flow.Identifier]*flow.Identity
+	lastStakingKey     crypto.PublicKey
+}
+
 // stakingKeysAggregator is a structure that aggregates the staking
 // public keys for QC verifications.
 type stakingKeysAggregator struct {
-	lastStakingSigners map[flow.Identifier]*flow.Identity
-	lastStakingKey     crypto.PublicKey
-	sync.RWMutex
+	current unsafe.Pointer // *aggregate type
+	inner   *aggregate
 }
 
 // creates a new staking keys aggregator
 func newStakingKeysAggregator() *stakingKeysAggregator {
-	aggregator := &stakingKeysAggregator{
+	aggregator := &stakingKeysAggregator{}
+
+	aggregator.inner = &aggregate{
 		lastStakingSigners: map[flow.Identifier]*flow.Identity{},
 		lastStakingKey:     NeutralBLSPublicKey(),
-		RWMutex:            sync.RWMutex{},
 	}
+
+	aggregator.current = unsafe.Pointer(aggregator.inner)
```

Comment:
I think `updateCurrent` would be better choice here, without inner

Reply:
I think I tried that and there was a problem in the tests.  If you want to paste your code change suggestion, I'll try it.